sklearn/utils/estimator_checks.py

```diff
@@ -554,6 +555,41 @@ def check_sample_weights_list(name, estimator_orig):
         estimator.fit(X, y, sample_weight=sample_weight)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_sample_weights_invariance(name, estimator_orig):
+    # check that the estimators yield same results for
+    # unit weights and no weights
+    if (has_fit_parameter(estimator_orig, "sample_weight") and
+            not (hasattr(estimator_orig, "_pairwise")
+                 and estimator_orig._pairwise) and
+            name not in ["KMeans", "MiniBatchKMeans"]):
+        # We skip pairwise because the data is not pairwise
+        # KMeans and MiniBatchKMeans were unstable; hence skipped.
```

Comment:
I'm confused about it. Are they still unstable under a fixed random_state. If so, is there a bug?
(Sorry if I made a mistake :) Go ahead if you have enough confidence)

Reply:
This is strange. When I initially ran this without using `set_random_state`, they were failing. After using `set_random_state`, they seem to be stable. I wonder if `estimator.fit(...,random_state=0)` is missing something that `set_random_state` does not. I can include those estimators to the test. @GaelVaroquaux WDYT?